Restructure and update repo (#31)
wormhole-foundation/wormhole-solidity-sdk: README.md

# Wormhole Solidity SDK

The purpose of this SDK is to make integrations with Wormhole on EVM compatible chains as smooth as possible by providing all necessary Solidity interfaces along with useful libraries and tools for testing.

This SDK was originally created for integrations with the WormholeRelayer and then expanded to cover all integrations.

## Installation

**Foundry and Forge**

```bash
forge install wormhole-foundation/wormhole-solidity-sdk
```

**Solc Version**

> Currently the SDK uses solc version 0.8.19 to avoid issues with OPCODE0 which was introduced in 0.8.20 but which is not supported on many EVM chains.

Reviewer: ```suggestion
Currently the SDK uses solc version 0.8.19 to avoid issues with PUSH0 which was introduced in 0.8.20 but which is not supported on many EVM chains.
```

Reviewer: should we use more permissive compiler version bounds (in the source files) and suggest instead to target `paris`?

Author: Yes, I [had to do that in the branch where I added openzeppelin](https://github.com/wormhole-foundation/wormhole-solidity-sdk/blob/9c3852e5ff83f6e5c6ec969c2ac5cd97b163eede/foundry.toml) because it is on 8.20 anyway...

Author: Strictly speaking, one should be going over individual files to determine their necessary pragma versions... some make use of solc features that were only introduced in the high teens while others are entirely unconstrained.

## WormholeRelayer

### Introduction

The WormholeRelayer (also sometimes referred to as the automatic or generic relayer) allows integrators to leverage external parties known as delivery providers, to relay messages emitted on a given source chain to the intended target chain.

> This frees integrators, who are building a cross-chain app, from the cumbersome and painful task of having to run relaying infrastructure themselves, without off-loading the burden of having to submit any transactions on the target chain themselves (and thus e.g. dealing with the headache of having to acquire gas tokens for the target chain).

Reviewer: ```suggestion
This frees integrators, who are building a cross-chain app, from the cumbersome and painful task of having to run relaying infrastructure themselves (and thus e.g. dealing with the headache of having to acquire gas tokens for the target chain).
```

Reviewer: even simpler:
```suggestion
This frees cross-chain app developers from having to run infrastructure that submits transactions on the destination chain (and thus e.g. having to maintain a gas token balance on that chain).
```

Author: If my writing was only half as elegant and concise as my code...
